src/ObjectIndexer.php

```diff
@@ -34,6 +34,25 @@ public function index(Indexable $object, string $type = "_doc")
         $this->index->index($params);
     }
 
+    /**
+     * @param Indexable[] $objects
+     * @param string $type
+     */
+    public function bulkIndex(array $objects, string $type = "_doc")
+    {
+        $params['type'] = $type;
+
+        foreach ($objects as $indexable) {
+            $key = $indexable->shouldBeIndexed() ? 'index' : 'delete';
```

Comment:
A type check for `Indexable` here would be great to raise a more relevant Exception

Reply:
agreed.

Would you be opposed to adding a dependency to https://github.com/webmozart/assert to handle that check ?

Reply:
I'm ok with it. The only risk I see is locking users to specific major version, but it's an easily reversible choice (and the lib seems to have very few  reason to introduce BC changes).

Reply:
done